gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/conn/ConnectionSignaller.java

```diff
@@ -239,6 +240,9 @@ public void run() {
           break;
         }
       }
+      // This clears the pending references of the offheap.
+      // Kept this call here to avoid creating a new thread. 
+      UnsafeHolder.releasePendingReferences();
```

Comment:
Thinking about it could be better just before FinalizeObject.getServerHolder().invokePendingFinalizers(). Reason is that this would be a short call whereas ours can be more expensive having to finalize statements/connections.